Simplified/Views/NYPLRoundedButton.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  NYPLRoundedButton.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2021-03-31.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import UIKit
+
+private let NYPLRoundedButtonPadding: CGFloat = 6.0
+
+@objc enum NYPLRoundedButtonType: Int {
+  case normal
+  case clock
+}
+
+@objc class NYPLRoundedButton: UIButton {
+  // Properties
+  private var type: NYPLRoundedButtonType {
+    didSet {
+      updateViews()
+    }
+  }
+  private var endDate: Date? {
+    didSet {
+      updateViews()
+    }
+  }
+  private var isFromDetailView: Bool
+  
+  // UI Components
+  private let label: UILabel = UILabel()
+  private let iconView: UIImageView = UIImageView()
+  
+  // Initializer
+  init(type: NYPLRoundedButtonType, endDate: Date?, isFromDetailView: Bool) {
+    self.type = type
+    self.endDate = endDate
+    self.isFromDetailView = isFromDetailView
+    
+    super.init(frame: CGRect.zero)
+    
+    setupUI()
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: - Setter
+  @objc func setType(_ type: NYPLRoundedButtonType) {
+    self.type = type
+  }
+  
+  @objc func setEndDate(_ date: NSDate?) {
+    guard let convertedDate = date as Date? else {
+      return
+    }
+    endDate = convertedDate
+  }
+  
+  @objc func setFromDetailView(_ isFromDetailView: Bool) {
+    self.isFromDetailView = isFromDetailView
+  }
+  
+  // MARK: - UI
+  private func setupUI() {
+    titleLabel?.font = UIFont.systemFont(ofSize: 14)
+    layer.borderColor = tintColor.cgColor
+    layer.borderWidth = 1
+    layer.cornerRadius = 3
+    
+    label.textColor = self.tintColor
+    label.font = UIFont.systemFont(ofSize: 9)
+    
+    addSubview(label)
+    addSubview(iconView)
+  }
+  
+  private func updateViews() {
+    let padX = NYPLRoundedButtonPadding + 2
+    let padY = NYPLRoundedButtonPadding
+    
+    if (self.type == .normal || self.isFromDetailView) {
+      self.contentEdgeInsets = isFromDetailView ? UIEdgeInsets(top: 8, left: 20, bottom: 8, right: 20) : UIEdgeInsets(top: padY, left: padX, bottom: padY, right: padX)
+      self.iconView.isHidden = true
+      self.label.isHidden = true
+    } else {
+      self.iconView.image = UIImage.init(named: "Clock")?.withRenderingMode(.alwaysTemplate)
+      self.iconView.isHidden = false
+      self.label.isHidden = false
+      if let endDate = self.endDate {
+        self.label.text = endDate.timeUntilString(suffixType: .short)
+      }
```

Comment:
I think you need an `else` case here, to cover the case where the UI goes from having a endDate to not having it... unless the label is hidden in some other way. You could also remove the `if` and do `endDate?.timeUntilString... ?? ""`

Reply:
Fixed